lib/templates/deploy.batch.go.template

```diff
@@ -0,0 +1,68 @@
+goto Deployment
+
+:: Utility Functions
+:: -----------------
+
+:SelectGoVersion
+
+IF DEFINED KUDU_SELECT_GO_VERSION_CMD (
+    :: TODO
+goto :EOF
+
+::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
+:: Deployment
+:: ----------
+
+:Deployment
+ECHO Handling Go deployment.
+
+:: 1. Prepare workspace
+IF "%GOROOT%" == "" (
+    :: select go version TODO
+    SET GOROOT=D:\Program Files\go\1.4.2
+)
+ECHO GOROOT %GOROOT%
+
+SET GOPATH=%DEPLOYMENT_TEMP%\gopath
+SET GOEXE="%GOROOT%\bin\go.exe"
+SET FOLDERNAME=azureapp
+SET GOAZUREAPP=%DEPLOYMENT_TEMP%\gopath\src\%FOLDERNAME%
+
+IF EXIST %GOPATH% (
+    ECHO %GOPATH% already exist
+    
+    ECHO Removing %GOAZUREAPP%
+    RMDIR /S /Q %GOAZUREAPP%
+) else (
+    ECHO Creating %GOPATH%\bin
+    MKDIR "%GOPATH%\bin"
+    
+    ECHO Creating %GOPATH%\pkg
+    MKDIR "%GOPATH%\pkg"
+    
+    ECHO Creating %GOPATH%\src
+    MKDIR "%GOPATH%\src"
+)
+
+ECHO Creating %GOAZUREAPP%
+MKDIR %GOAZUREAPP%
+
+:: 2. KuduSync
+ECHO Prepare source files for build
+IF /I "%IN_PLACE_DEPLOYMENT%" NEQ "1" (
+  call :ExecuteCmd "%KUDU_SYNC_CMD%" -v 50 -f "%DEPLOYMENT_SOURCE%" -t "%GOAZUREAPP%" -n "%NEXT_MANIFEST_PATH%" -p "%PREVIOUS_MANIFEST_PATH%" -i ".git;.hg;.deployment;deploy.cmd"
+  IF !ERRORLEVEL! NEQ 0 goto error
+)
+
+:: 3. Build
+ECHO Copy static files to deployment
+call :ExecuteCmd "%KUDU_SYNC_CMD%" -v 50 -f "%GOAZUREAPP%" -t "%DEPLOYMENT_TARGET%" -n "%NEXT_MANIFEST_PATH%-deploy" -p "%PREVIOUS_MANIFEST_PATH%-deploy" -i ".git;.hg;.deployment;deploy.cmd;*.go"
```

Comment:
this kudusync invacation should use the usual manifest path the previous one should use a non-regular one.
(think of a scenario where you deployed go and then node app...)